func_keys.go

```diff
@@ -0,0 +1,106 @@
+package dasel
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+type ErrInvalidType struct {
+	ExpectedTypes []string
+	CurrentType   string
+}
+
+func (e *ErrInvalidType) Error() string {
+	return fmt.Sprintf("unexpected types: expect %s, get %s", strings.Join(e.ExpectedTypes, " "), e.CurrentType)
+}
+
+func (e *ErrInvalidType) Is(other error) bool {
+	o, ok := other.(*ErrInvalidType)
+	if !ok {
+		return false
+	}
+	if len(e.ExpectedTypes) != len(o.ExpectedTypes) {
+		return false
+	}
+	if e.CurrentType != o.CurrentType {
+		return false
+	}
+	for i, t := range e.ExpectedTypes {
+		if t != o.ExpectedTypes[i] {
+			return false
+		}
+	}
+	return true
+}
+
+var KeysFunc = BasicFunction{
+	name: "keys",
+	runFn: func(c *Context, s *Step, args []string) (Values, error) {
+		if err := requireNoArgs("keys", args); err != nil {
+			return nil, err
+		}
+
+		input := s.inputs()
+
+		res := make(Values, len(input))
+
+		for i, val := range input {
+			switch val.Kind() {
+			case reflect.Slice, reflect.Array:
+				list := make([]any, 0, val.Len())
+
+				for i := 0; i < val.Len(); i++ {
+					list = append(list, i)
+				}
+
+				res[i] = ValueOf(list)
+			case reflect.Map:
+				keys := val.Unpack().MapKeys()
```

Comment:
```suggestion
				keys := val.MapKeys()
```

I'm pretty sure you can go straight to `MapKeys()` here thanks to https://github.com/TomWright/dasel/blob/88bf4414bc0b906034146a0ee2dd005f05069418/value.go#L164